File: leetcode/js/ReverseString.js
//IMPORTANT!! Submit Code Region Begin(Do not remove this line)
/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 *
 *  This method uses TWO-POINTER APPROACH
 *
 */

// Two-pointer Approach
const reverseString = function (s) {
    let left = 0, right = s.length - 1;
    while (left < right) {

        let temp = s[left];
        s[left] = s[right];
        s[right] = temp

        left++;
        right--;
    }
    return s;
};
//IMPORTANT!! Submit Code Region End(Do not remove this line)

// Example usage:
const test = "hello";
const output = reverseString(test.split("")).join("");
console.log(output); // Output: ["o", "l", "l", "e", "h"]

File: leetcode/JAVA/src/main/java/leetcodecn/ReverseString.java
package leetcodecn;

class ReverseString {

    //IMPORTANT!! Submit Code Region Begin(Do not remove this line)
    class Solution {
        public void reverseString(char[] s) {
            int left = 0;
            int right = s.length - 1;
            while (left < right) {
                char temp = s[left];
                s[left] = s[right];
                s[right] = temp;
                left++;
                right--;
            }
        }
    }
//IMPORTANT!! Submit Code Region End(Do not remove this line)

    public static void main(String[] args) {
        // test reverseString
        ReverseString rs = new ReverseString();
        Solution solution = rs.new Solution();
        char[] s = {'h', 'e', 'l', 'l', 'o'};
        solution.reverseString(s);
        System.out.println(s); // should print "olleh"
    }
}